k9mail/src/main/java/com/fsck/k9/Account.java

```diff
@@ -820,6 +820,45 @@ public AccountStats getStats(Context context) throws MessagingException {
         return stats;
     }
 
+    public int getFolderUnreadCount(Context context, String folderName) throws MessagingException {
+        if (!isAvailable(context)) {
+            return 0;
+        }
+
+        int unreadMessageCount = 0;
+        ContentResolver cr = context.getContentResolver();
+
+        Uri uri = Uri.withAppendedPath(EmailProvider.CONTENT_URI,
+                "account/" + getUuid() + "/stats");
+
+        String[] projection = {
+                StatsColumns.UNREAD_COUNT,
+        };
+
+        LocalSearch search = new LocalSearch();
```

Comment:
replace this whole block with a function that explains intent:

`buildUnreadQueryForFolder(folderName)`